src/hotspot/share/compiler/compilerDefinitions.cpp

```diff
@@ -136,10 +136,12 @@ intx CompilerConfig::scaled_compile_threshold(intx threshold, double scale) {
     }
     int exp;
     (void) frexp(v, &exp);
-    if (exp > 63) {
+    if (exp > LP64_ONLY(63) NOT_LP64(31)) {
```

Comment:
How about `(exp > (sizeof(intx)*BitsPerByte-1))`?

Reply:
`(exp > BitsPerWord-1)` then?

Reply:
Right. See the commit, I introduced a variable to implicitly cast to `int` and avoid signed-unsigned comparison.